charts/super-agent/templates/_helm-release.yaml

```diff
@@ -0,0 +1,42 @@
+{{- define "newrelic-super-agent.helm.release" -}}
+{{- $releaseName := include "newrelic.common.naming.truncateToDNSWithSuffix" ( dict "name" (include "newrelic.common.naming.fullname" .) "suffix" "deployment" ) -}}
+{{- /*
+To understand why this manifest is a template function instead of a manifest that is appyed directly, read the comment at `job-manifests.yaml`.
+*/ -}}
+apiVersion: helm.toolkit.fluxcd.io/v2beta1
+kind: HelmRelease
+metadata:
+  labels:
+    {{- include "newrelic.common.labels" . | nindent 4 }}
+  name: {{ $releaseName }}
+  namespace: {{ .Release.Namespace }}
+spec:
+  interval: {{ .Values.helm.repository.updateInterval }}
+  chart:
+    spec:
+      chart: {{ .Values.helm.release.chart }}
+      version: {{ .Values.helm.release.version }}
+      sourceRef:
+        kind: HelmRepository
+        name: newrelic
+        namespace: {{ .Release.Namespace }}
+      interval: {{ .Values.helm.repository.updateInterval }}
+  values:
+    fullnameOverride: super-agent
+    {{- with .Values.helm.release.values }}
+    {{- . | toYaml | nindent 4 }}
+    {{- end }}
+  {{- with .Values.helm.release.valuesFrom }}
+  valuesFrom:
```

Comment:
are changes on this values retriggering a redeploy of the SA? do you know how much time is needed to see the effects?

Reply:
Really good question, it took me some time to figure out.

There is an issue that says that they planned to have watch capabilities that never implemented: https://github.com/fluxcd/helm-controller/issues/42

So, the way to trigger a release is by using `interval`.

There are two intervals:
 * [Helm releases' `.spec.interval`](https://github.com/newrelic/helm-charts/pull/1175/files#diff-74e915ed23d7fc934887bc2a55f84ef93d8ca1e7d0c6a3e235be39830a6532d9R14) spscifies the reconcile interval so it would take `.spec.interval` to set the release with the new values.
 * [Helm releases' `.spec.chart.spec.interval`](https://github.com/newrelic/helm-charts/pull/1175/files#diff-74e915ed23d7fc934887bc2a55f84ef93d8ca1e7d0c6a3e235be39830a6532d9R23) checks for chart updates.
 * [Helm repository' `.spec.interval`](https://github.com/newrelic/helm-charts/pull/1175/files#diff-470e71a1fb8171edfe1dc2f4493e4b25fe730119f40ca39ee905d680d459521eR13) set the amount of time the source controller publishes a new chart for the helm controller to set it up.
 
Do you think it makes sense to have different intervals? I reused the same value ([this one](https://github.com/newrelic/helm-charts/pull/1175/files#diff-3bd49393f47bdfcb35b90c33237f384109da730bc892fa317ba8db58d29d7975R10)) for all of them.

Reply:
I would say that .spec.interval should be as faster as it could to make changes to values be applied as fast as possible. 
i think the other two we can tackled in the future when autoupdates are analysed